zscaler/package.yaml

```
@@ -0,0 +1,217 @@
+
+      // === Classification ===
+      // Map the request method to the OCSF activity ID and name.
+      ocsf.activity_id = $activity_map[zia.requestmethod]? else 0
+      move ocsf.activity_name = zia.requestmethod
+
+      // Set the static class and category for HTTP Activity.
+      ocsf.category_uid = 4
+      ocsf.category_name = "Network Activity"
+      ocsf.class_uid = 4002
+      ocsf.class_name = "HTTP Activity"
+
+      // Derive the specific event type from the class and activity.
+      ocsf.type_uid = ocsf.class_uid * 1000 + ocsf.activity_id
+      ocsf.type_name = ocsf.activity_name
+
+      // Map the Zscaler action to OCSF disposition.
+      ocsf.disposition_id = $disposition_map[zia.action]? else 0
+      move ocsf.disposition = zia.action
+      move ocsf.status_detail = zia.reason
+
+      // Determine status based on the HTTP response code.
+      // Codes in the 2xx and 3xx range are generally successful.
+      if zia.status == "NA" {
+        status_code_num = null
+      } else {
+        status_code_num = int(zia.status)
+      }
+      if status_code_num == null {
+        ocsf.status = "Unknown"
+        ocsf.status_id = 0
+      } else if status_code_num >= 200 and status_code_num < 400 {
+        ocsf.status = "Success"
+        ocsf.status_id = 1
+      } else if status_code_num >= 400 {
+        ocsf.status = "Failure"
+        ocsf.status_id = 2
+      } else {
+        ocsf.status = "Unknown"
+        ocsf.status_id = 0
+      }
+
+      // Map threat severity to OCSF severity.
+      if zia.threatseverity == "None" {
+        ocsf.malware.severity_id = 0
+        ocsf.malware.severity = "Unknown"
+      } else if zia.threatseverity == "medium" {
+        ocsf.malware.severity_id = 3
+        ocsf.malware.severity = "Medium"
+      } else if zia.threatseverity == "critical" {
+        ocsf.malware.severity_id = 5
+        ocsf.malware.severity = "Critical"
+      } else {
+        ocsf.malware.severity_id = 99
+        ocsf.malware.severity = zia.threatseverity
+      }
+      //drop zia.threatseverity
+
+      // Map the remaining theat* fields to OCSF malware.
+      if zia.threatclass == "None" {
+        zia.threatclass = null
+      }
+      if zia.threatcategory == "None" {
+        zia.threatcategory = null
+      }
+      ocsf.malware = [{
+        name: move zia.threatname,
+        classifications: [move zia.threatclass?, move zia.threatcategory?]
+      }]
+
+      // === Occurrence ===
+      // Parse the zia datetime string into a valid timestamp.
+      // The format string '%a %b %d %H:%M:%S %Y' matches "Tue Jul 22 10:59:31 2025".
+      ocsf.time = parse_time(zia.datetime, "%a %b %d %H:%M:%S %Y")
+      drop zia.datetime
+
+      // === Context ===
+      ocsf.metadata = {
+        log_name: "NSS",
+        product: {
+          name: move zia.product,
+          vendor_name: move zia.vendor,
+        },
+        uid: move zia.event_id,
+        version: "1.5.0", // The OCSF schema version.
```

```suggestion
        version: "1.5.0",
```

There are numerous comments in here similar to this that we can easily drop. This helps creating more of a visible pipeline structure when reviewing it.